src/aliceVision/keyframe/KeyframeSelector.cpp

```diff
@@ -25,452 +25,711 @@ namespace keyframe {
-  : _mediaPaths(mediaPaths)
-  , _sensorDbPath(sensorDbPath)
-  , _voctreeFilePath(voctreeFilePath)
-  , _outputFolder(outputFolder)
+    : _mediaPaths(mediaPaths)
+    , _sensorDbPath(sensorDbPath)
+    , _outputFolder(outputFolder)
 {
-  if((_maxOutFrame != 0) &&
-     !_hasSharpnessSelection &&
-     !_hasSparseDistanceSelection)
-  {
-      ALICEVISION_LOG_ERROR("KeyframeSelector needs at least one selection method if output frame limited !");
-      throw std::invalid_argument("KeyframeSelector needs at least one selection method if output frame limited !");
-  }
-
-  // load vocabulary tree
-  _voctree.reset(new aliceVision::voctree::VocabularyTree<DescriptorFloat>(voctreeFilePath));
-
-  {
-      ALICEVISION_LOG_INFO("vocabulary tree loaded with :" << std::endl
-                       << "\t- " << _voctree->levels() << " levels" << std::endl
-                       << "\t- " << _voctree->splits() << " branching factor" << std::endl);
-  }
-
-  // check number of input media filePaths
-  if(mediaPaths.empty())
-  {
-    ALICEVISION_LOG_ERROR("Cannot create KeyframeSelector without a media file path !");
-    throw std::invalid_argument("Cannot create KeyframeSelector without a media file path !");
-  }
-
-  // resize mediasInfo container
-  _mediasInfo.resize(mediaPaths.size());
-
-  // create SIFT image describer
-  _imageDescriber.reset(new feature::ImageDescriber_SIFT());
+    // Check that a least one media file path has been provided
+    if (mediaPaths.empty()) {
+        ALICEVISION_LOG_ERROR("Cannot create KeyframeSelector without at least one media file path!");
+        throw std::invalid_argument("Cannot create KeyframeSelector without at least one media file path!");
+    }
+
+    scoresMap["Sharpness"] = &_sharpnessScores;
+    scoresMap["OpticalFlow"] = &_flowScores;
 }
 
-void KeyframeSelector::process()
+void KeyframeSelector::processRegular()
 {
-  // create feeds and count minimum number of frames
-  std::size_t nbFrames = std::numeric_limits<std::size_t>::max();
-  for(std::size_t mediaIndex = 0; mediaIndex < _mediaPaths.size(); ++mediaIndex)
-  {
-    const auto& path = _mediaPaths.at(mediaIndex);
-
-    // create a feed provider per mediaPaths
-    _feeds.emplace_back(new dataio::FeedProvider(path));
-
-    const auto& feed = *_feeds.back();
-
-    // check if feed is initialized
-    if(!feed.isInit())
-    {
-      ALICEVISION_LOG_ERROR("Cannot initialize the FeedProvider with " << path);
-      throw std::invalid_argument("Cannot while initialize the FeedProvider with " + path);
-    }
+    _selectedKeyframes.clear();
+
+    std::size_t nbFrames = std::numeric_limits<unsigned int>::max();
+    std::vector<std::unique_ptr<dataio::FeedProvider>> feeds;
+
+    for (std::size_t mediaIndex = 0; mediaIndex < _mediaPaths.size(); ++mediaIndex) {
+        const auto& path = _mediaPaths.at(mediaIndex);
+
+        // Create a feed provider per mediaPaths
+        feeds.emplace_back(new dataio::FeedProvider(path));
```

Comment:
it should be safer to use `make_unique()` https://stackoverflow.com/a/15784813  @fabiencastan 
```suggestion
        feeds.push_back(std::make_unique<dataio::FeedProvider>(path));
```